pageserver/src/http/routes.rs

```diff
@@ -324,6 +324,10 @@ impl From<crate::tenant::TimelineArchivalError> for ApiError {
         match value {
             NotFound => ApiError::NotFound(anyhow::anyhow!("timeline not found").into()),
             Timeout => ApiError::Timeout("hit pageserver internal timeout".into()),
+            HasArchivedParent(parent) => ApiError::PreconditionFailed(
+                format!("Cannot unarchive timeline which has archived ancestor: {parent:?}")
+                    .into_boxed_str(),
```

Comment:
```suggestion
            e @ HasArchivedParent(_) => ApiError::PreconditionFailed(
                format!("{e:}")
                    .into_boxed_str(),
```

this stringification is no longer needed, right?

Reply:
this string is expected in the added test.

Reply:
it isn't but I'd be rather cautious about using error formatting directly in `ApiError` conversions: we will expose the ApiError to users directly, and any obfuscation how the error is formatted could accidentally expose that info.

Reply:
Also it would be inconsistent with the other errors in the match

Reply:
ok seems that `AlreadyInProgress` also has it, so I just did it